### PR TITLE
refactor(ApiGateway): Make HttpClient overridable in RequestInfo #NP-47966

### PR DESCRIPTION
**Pull request by torbjokv**
https://sikt.atlassian.net/browse/NP-47966

### CODE REVIEW

File: apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java [hunk 1]
```diff
@@ -100,29 +102,36 @@ public RequestInfo(HttpClient httpClient, Supplier<URI> cognitoUri, Supplier<URI
...
 
-    public static RequestInfo fromRequest(InputStream requestStream) {
-        return attempt(() -> JsonUtils.dtoObjectMapper.readValue(requestStream, RequestInfo.class)).orElseThrow();
+    public static RequestInfo fromRequest(InputStream requestStream, HttpClient httpClient) throws ApiIoException {
+        String inputString = IoUtils.streamToString(requestStream);
+        return fromString(inputString, httpClient);
+    }
+
+    public static RequestInfo fromString(String inputString, HttpClient httpClient) throws ApiIoException {
+        var requestInfo = new ApiMessageParser<>(mapper).getRequestInfo(inputString);
```
Comment (torbjokv) [inline apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java:123]: Unsure if it's any point exposing mapper outside this lib or not, probably not?. would be nice if we could avoid having it in our constructors all over the place

File: apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java [hunk 2]
```diff
@@ -71,9 +73,9 @@
 public class RequestInfo {
 
     public static final String ERROR_FETCHING_COGNITO_INFO = "Could not fetch user information from Cognito:{}";
-    private static final HttpClient DEFAULT_HTTP_CLIENT = HttpClient.newBuilder().build();
```
Comment (torbjokv) [inline apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java:74]: removing this is the main motivation with the PR

File: apigateway/src/test/java/nva/commons/apigateway/RequestInfoTest.java [hunk 3]
```diff
@@ -303,60 +305,77 @@ void shouldReturnNotAuthorizedWhenRequestInfoDoesNotContainAccessRightsAndCheckI
     }
 
     @Test
-    void shouldReturnMultiValueQueryValuesWhenProvided() throws JsonProcessingException {
+    void shouldReturnReturnBadRequestWhenObjectMapperFails(){
```
Comment (truhacevkir) [inline apigateway/src/test/java/nva/commons/apigateway/RequestInfoTest.java:308]: *ReturnReturn

**Review by truhacevkir — APPROVED** [review]

File: apigateway/src/main/java/nva/commons/apigateway/ApiGatewayHandler.java [hunk 4]
```diff
@@ -51,23 +52,16 @@ public abstract class ApiGatewayHandler<I, O> extends RestRequestHandler<I, O> {
...
 
     private Supplier<Map<String, String>> additionalSuccessHeadersSupplier;
     private boolean isBase64Encoded;
 
     public ApiGatewayHandler(Class<I> iclass) {
-        this(iclass, new Environment());
+        this(iclass, new Environment(), defaultRestObjectMapper, HttpClient.newBuilder().build());
     }
 
-    public ApiGatewayHandler(Class<I> iclass, Environment environment) {
```
Comment (torbjokv) [inline apigateway/src/main/java/nva/commons/apigateway/ApiGatewayHandler.java:63]: hmm not sure if it was smart to remove this. This causes me to manually set JsonUtils.dtoObjectMapper in constructors in publication-api. add it back?

```java
  public CreatePublicationHandler(ResourceService publicationService,
                                    Environment environment,
                                    IdentityServiceClient identityServiceClient,
                                    SecretsManagerClient secretsManagerClient,
                                    HttpClient httpClient) {
        super(CreatePublicationRequest.class, environment, JsonUtils.dtoObjectMapper, httpClient);
```

Comment (torbjokv) [reply apigateway/src/main/java/nva/commons/apigateway/ApiGatewayHandler.java:63]: removed mapper from constructor, hopefully not needed anywhere. not in publication-api at least

File: apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java (hunk 2 above)
Comment (LarsV123) [reply apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java:74]: Can you add a link to the Jira ticket and/or a PR description to explain what the PR is intended to do?

Comment (torbjokv) [reply apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java:74]: image
there it is :-P but yes, I should add link as minimum at least

File: apigateway/src/main/java/nva/commons/apigateway/ApiGatewayHandler.java (hunk 4 above)
Comment (LarsV123) [reply apigateway/src/main/java/nva/commons/apigateway/ApiGatewayHandler.java:63]: Are you sure this change is compatible with all projects that use this?

Perhaps you could keep the old constructor temporarily and mark it as deprecated, until all dependent projects are updated to use this new constructor?

Comment (torbjokv) [reply apigateway/src/main/java/nva/commons/apigateway/ApiGatewayHandler.java:63]: its not compatible as in it is a breaking change in the constructor, but small and simple changes, so I don't see a big issue fixing it "on the fly".

**Review by LarsV123 — APPROVED** [review]
lgtm (lets gamble, try merging)

**Pull request merged**